library/java/org/chromium/net/impl/CancelProofEnvoyStream.java

```diff
@@ -7,48 +7,58 @@
 import io.envoyproxy.envoymobile.engine.EnvoyHTTPStream;
 
 /**
- * Consistency layer above the {@link EnvoyHTTPStream} preventing unwarranted Stream operations
- * after a "cancel" operation. There are no "synchronized" - this is Compare And Swap based logic.
+ * CancelProofEnvoyStream is a consistency layer above the {@link EnvoyHTTPStream} preventing
+ * unwarranted Stream operations after a "cancel" operation. There are no "synchronized" - this is
+ * Compare And Swap based logic. This class is Thread Safe.
  *
  * <p>This contraption ensures that once a "cancel" operation is invoked, there will be no further
  * operations allowed with the EnvoyHTTPStream - subsequent operations will be ignored silently.
  * However, in the event that that one or more EnvoyHTTPStream operations are currently being
  * executed, the "cancel" operation gets postponed: the last concurrent operation will invoke
  * "cancel" at the end.
  *
- * <p>Instance of this class start with a state of "Busy Starting". This ensure that if a cancel
+ * <p>Instances of this class start with a state of "Busy Starting". This ensure that if a cancel
  * is invoked while the stream is being created, that cancel will be executed only once the stream
  * is completely initialized. Doing otherwise leads to unpredictable outcomes.
  */
 final class CancelProofEnvoyStream {
 
   private static final int CANCEL_BIT = 0x8000;
+  /**
+   * mState mainly maintains a counter of how many Stream operations are currently in-flight.
+   * However, when 15 is flipped (0x8000), it indicates that cancel operation hase been requested.
```

Comment:
nit: "However when bit 15 (0x80000) is set, it indicates that the cancel operation has been requested"

Reply:
Done, thanks.